### ISSUE TITLE
0.5.7 context flag error

### ISSUE THREAD

**Issue by dubcanada** [ariga/terraform-provider-atlas#99]
Hello,

I just upgraded from 0.5.6 to 0.5.7 and I am getting an error

```
│ Error: Failed to apply migrations
│
│   with atlas_migration.backend-migration,
│   on rds.tf line 73, in resource "atlas_migration" "backend-migration":
│   73: resource "atlas_migration" "backend-migration" {
│
│ Error: unknown flag: --context
```

Downgrading to 0.5.6 does fix it.

Is there any additional upgrading that needs to be done?

Thanks,

**Comment by rotemtam** [comment]
Try upgrading Atlas to the latest version:


> curl -sSf https://atlasgo.sh | sh

**Comment by giautm** [comment]
Hello @dubcanada, yes. That's commit add the context argument to the CLI and it happen after the version v0.15.0 was released.

But, the terraform-provider-atlas used an atlas binary that bundled with it during build process. So, you can extract the provider from our release page and see there is a binary in here. https://github.com/ariga/terraform-provider-atlas/releases/tag/v0.5.7 And that version is canary build, and it has the context argument.

What's the OS you're using?

**Comment by dubcanada** [comment]
pop_os 22.04 (jammy) let me try a few things, I will confirm if I am still experiencing it after with further details, could just be a me problem

**Comment by giautm** [comment]
Hello @dubcanada, I'm also continuing to investigate, I'm suspecting the atlas file is missing execution permissions. So TF-provider used the binary in `/usr/local/bin`.

**Comment by giautm** [comment]
Hello! After I spend few hours to [add integration test](https://github.com/ariga/terraform-provider-atlas/pull/101) for the provider. I surely believe that [the permissions of `atlas` binary is correct](https://github.com/ariga/terraform-provider-atlas/pull/100#issuecomment-1804292444) and it works as expected.

I still unable to find why you facing with this issue, if you have any clue. Please let me know!

**Comment by giautm** [comment]
Hi @dubcanada, I found the issue and fix with #102. Please upgrade the provider to v0.5.8 and try again.

Thank you so much!
